### ISSUE TITLE
Code Action createtypestub fails when executed as code action.

### ISSUE THREAD

**Issue by slyboots** [fannheyward/coc-pyright#391]
### note: Im currently at work and will flesh out this issue later this evening. I just wanted to get the issue in with some context since its fresh in my mind.

When my cursor is on an import statement for a module with out typestubs, the code action ui makes it seem like I can execute the action to create typestubs for the specific module.

![image](https://user-images.githubusercontent.com/22822605/111537682-bab14f00-8739-11eb-939b-e797b6491e07.png)
(_in the image above the code action ui displays a message asking if I would like to create pymongo typestubs. Which I do because typechecking_)

However, when I hit enter to execute the code action, the typestub creation fails with the following error: (_taken from output://Pyright_)  
```
[Error  - 3:45:26 PM] Import '/Users/dakotalorance/projects/pulley/lib/db.py' could not be resolved
[Trace - 3:45:26 PM] Received notification 'window/logMessage'.
Params: {
    "type": 1,
    "message": "An error occurred when creating type stub for '/Users/dakotalorance/projects/pulley/lib/db.py': Import '/Users/dakotalorance/projects/pulley/lib/db.py' could not be resolved"
}
```

It seems like the full path of the active file was provided for importing instead of the symbol that I was expecting ("pymongo").

Interestingly enough, if I execute createtypestubs via CocCommand there are no errors and the typestubs are created for the symbol I expected. e.g. `:CocCommand pyright.createtypestub pymongo<CR>` works perfectly.

This difference in behavior makes me think there is something weird going on with the way the arguments are being provided to the pyright server and [this code from the index.ts seems to reaffirm that](https://github.com/fannheyward/coc-pyright/blob/master/src/index.ts#L180)

```
        if (args[args.length - 1] !== filePath) {
          // args from Pyright   : [root, module, filePath]
          // args from CocCommand: [module]
          args.unshift(workspace.root);
          args.push(filePath);
        }
```

I dont have a lot of experience at writing language-server clients though, and have spent very little time looking at this projects code, so I could be completely off-base ¯\\\_(ツ)_/¯

I'm going to try and get a PR up for this unless someone else familiar with the project can fix it sooner.

**Comment by fannheyward** [comment]
Can't reproduce, what's your coc-pyright version? What `:CocCommand pyright.version` output?

**Comment by slyboots** [comment]
Output of `:CocCommand pyright.version`:  
`[coc.nvim] Pyright version 1.1.121`

Also apologize for the delay! I've never gotten a response so quickly to an issue before so I wasn't watching my notifications!
I do appreciate you taking a look at this though!

**Comment by yaegassy** [comment]
@slyboots I tried this in my environment, but `"5. Create Type Stub For "pymongo"` itself does not show up in the first place.

<img width="611" alt="coc-pyright-issue-391-1" src="https://user-images.githubusercontent.com/188642/111851108-70a4b500-8955-11eb-9cb7-3580fe26c326.png">

Can't reproduce, what's your `pymongo` version?

**Comment by slyboots** [comment]
@yaegassy thanks for checking this!

```
❯ pip list | grep pymongo
pymongo                   3.11.3
```

I think that the reason that you're not seeing the option to create a typestub file is because the pyright server isnt able to resolve the import for pymongo. When I uninstall pymongo the option to create a typestub is no longer listed.

However this issue isnt specific to pymongo. Im able to reproduce with other modules as well.

**Comment by yaegassy** [comment]
@slyboots The installation and import of the pymongo package in my environment is fine. The completion also works.

<img width="510" alt="coc-pyright-issue-391-2" src="https://user-images.githubusercontent.com/188642/111854854-c6348e00-8964-11eb-9452-e6da8ec065ba.png">

I was wondering if @fannheyward might not be able to reproduce this in pymongo as well as I can.

> However this issue isnt specific to pymongo. Im able to reproduce with other modules as well.

What is a simple case that I can recreate? https://github.com/fannheyward/coc-pyright#my-workflow-with-pyright

**Comment by slyboots** [comment]
>@slyboots The installation and import of the pymongo package in my environment is fine. The completion also works.
Weird... Im only able to get the behavior shown in your screenshot if I delete pymongo... 🤔 

I will be back with a simple test case script for you in just a bit!

**Comment by fannheyward** [comment]
@yaegassy put your cursor on `pymongo` you will get the create code action. 

I can get the action and it works as expected, the typings are created successfully.

**Comment by slyboots** [comment]
Quick update:
I just upgraded CoC (_I probably should have done that before writing this issue tbh_) and Im no longer able to get the create typestub action to appear.

Which I guess means this issue is fixed? A menu item cant be broken if it isnt there ¯\_(ツ)_/¯

**Comment by yaegassy** [comment]
OK, If fannheyward tries it and it behaves correctly, it's fine. 🙇 (For some reason, "Create Type Stub For..." doesn't show up in my environment, but...)

**Comment by fannheyward** [comment]
@slyboots can you create stub with this:

`node ~/.configure/coc/extensions/node-modules/coc-pyright/bin/pyright --createstub pumongo`?

Note: I'm posting on Mobile, the path maybe wrong, please give it a check.

**Comment by fannheyward** [comment]
I just found that you've installed Pyright globally? coc-pyright will install Pyright, you don't need to install will npm g.

**Comment by slyboots** [comment]
@fannheyward Thanks for pointing out the installation issue. I've uninstalled the global pyright from my system.

I still get the error when attempting to install typestubs via the CocAction though.

When I run: `node ~/.config/coc/extensions/node_modules/coc-pyright/node_modules/.bin/pyright --createstub pymongo`

**Comment by slyboots** [comment]
Separate from all of that, when I run `:CocCommand pyright.createTypeStub pymongo`

this is whats shown in the output:

```
[Trace - 9:14:53 PM] Sending request 'workspace/executeCommand - (330)'.
Params: {
    "command": "pyright.createtypestub",
    "arguments": [
        "/Users/dakotalorance/projects/gansu",
        "pymongo"
    ]
}
```

**Comment by slyboots** [comment]
@fannheyward I decided to rm everything related to coc, then reinstalled coc and coc-pyright.

Its working correctly now. 😞 🤦  I guess something wasnt correctly updated, but I should have tried this before making an issue.

I apologize for the wasted time.

**Comment by fannheyward** [comment]
You're using an old release of coc-pyright.

There're two ways to create type stub: code action provided by Pyright and `CocCommand pyright.createTypeStub` manually. They both run `workspace/executeCommand` request.

The different is: code action came from Pyright with three params: `[project root, module name, current file]`, `pyright.createTypeStub` manually has only module name. When runs `workspace/executeCommand`, coc-pyright will add project root to the params, after this, the first becomes `[project root, project root, module name, current file]`, this will make create stub failed. The second is `[project root, module name]`, this can create typings.

I've made a fix in coc-pyright and released in v1.1.122.

I'll improve `pyright.version` to output coc-pyright version too.
